src/gsi/control2state.f90

```diff
@@ -403,7 +404,15 @@ subroutine control2state(xhat,sval,bval)
    end if
 
 !  Same one-to-one map for chemistry-vars; take care of them together 
+   if(.not.laeroana_fv3cmaq .and. icvt_cmaq_fv3.eq.2)then
```

Comment:
If a few key options like laeroana_gocart, laeroana_fv3cmaq and icvt_cmaq_fv3 and maybe others could be explained in the gsimod.f90 or their corresponding declaration modules, that would be significantly improve the readability of this new function.